ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -671,53 +678,28 @@ class JITModBuilder {
 
     // Create a vector in which to store the locations of the corresponding
     // AOT variables.
-    StructType *AOTLocTy =
-        StructType::get(Context, {PointerSizedIntTy, PointerSizedIntTy,
-                                  Int8PtrTy, PointerSizedIntTy});
-    AllocaInst *AOTLocVec = FailBuilder.CreateAlloca(
-        AOTLocTy, ConstantInt::get(PointerSizedIntTy, LiveValues.size()));
-    std::map<std::string, Value *> FuncPtrMap;
+    AOTInfo *AOTMappingVec = (AOTInfo *) malloc(sizeof(AOTInfo) * LiveValues.size());
     for (size_t I = 0; I < LiveValues.size(); I++) {
       Value *Live = LiveValues[I];
       std::tuple<size_t, size_t, Instruction *, size_t> Entry = AOTMap[Live];
       size_t BBIdx = std::get<0>(Entry);
       size_t InstrIdx = std::get<1>(Entry);
       Instruction *AOTVar = std::get<2>(Entry);
       size_t StackFrameIdx = std::get<3>(Entry);
-      auto iter = FuncPtrMap.find(AOTVar->getFunction()->getName().data());
-      Value *FPtr;
-      if (iter == FuncPtrMap.end()) {
-        // FIXME: Use function index instead of string name.
-        FPtr =
-            FailBuilder.CreateGlobalStringPtr(AOTVar->getFunction()->getName());
-        FuncPtrMap.insert({AOTVar->getFunction()->getName().data(), FPtr});
-      } else {
-        FPtr = iter->second;
-      }
-      auto GEP = FailBuilder.CreateGEP(AOTLocTy, AOTLocVec,
-                                       {ConstantInt::get(PointerSizedIntTy, I),
-                                        ConstantInt::get(Int32Ty, 0)});
-      FailBuilder.CreateStore(ConstantInt::get(PointerSizedIntTy, BBIdx), GEP);
-      GEP = FailBuilder.CreateGEP(AOTLocTy, AOTLocVec,
-                                  {ConstantInt::get(PointerSizedIntTy, I),
-                                   ConstantInt::get(Int32Ty, 1)});
-      FailBuilder.CreateStore(ConstantInt::get(PointerSizedIntTy, InstrIdx),
-                              GEP);
-      GEP = FailBuilder.CreateGEP(AOTLocTy, AOTLocVec,
-                                  {ConstantInt::get(PointerSizedIntTy, I),
-                                   ConstantInt::get(Int32Ty, 2)});
-      FailBuilder.CreateStore(FPtr, GEP);
-      GEP = FailBuilder.CreateGEP(AOTLocTy, AOTLocVec,
-                                  {ConstantInt::get(PointerSizedIntTy, I),
-                                   ConstantInt::get(Int32Ty, 3)});
-      FailBuilder.CreateStore(
-          ConstantInt::get(PointerSizedIntTy, StackFrameIdx), GEP);
+      const char* FName = AOTVar->getFunction()->getName().data();
+
+      AOTInfo AI;
```

Comment:
Would it be neater to use an initializer instead of lots of assignments?

```
AOTInfo AI  = { BBIdx, InstrIdx, ...}
```

Reply:
I tried that first, but the compiler didn't like it. I can try and have another go.